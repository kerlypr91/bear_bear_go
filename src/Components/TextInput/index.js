import React from 'react'
import { Input } from 'semantic-ui-react'

export default function TextInput (props) {
  const { onChange, value, loading, placeholder, className } = props

  const handleChange = (event, data) => {
    const { value: newValue } = data

    onChange(newValue)
  }

  return (
    <Input
      inverted
      icon='search'
      value={value}
      onChange={handleChange}
      className={className}
      placeholder={placeholder}
      loading={loading}
    />
  )
}