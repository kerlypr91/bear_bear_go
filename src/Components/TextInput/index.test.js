import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './index';

test('Dropdown match snapshot', () => {
  const onChangeSpy = jest.fn()

  const props = {
    placeholder: 'test',
    loading: false,
    className: '',
    onChange: onChangeSpy,
    value: 'a'
  }

  const component = renderer.create(
    <TextInput {...props} />
  )

  // base snapshot

  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()

  // onChange prop gets called when a different text value is typed

  const input = tree.children[0]

  input.props.onChange()

  expect(onChangeSpy).toHaveBeenCalled()
});