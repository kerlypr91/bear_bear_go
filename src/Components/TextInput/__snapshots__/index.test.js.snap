// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown match snapshot 1`] = `
<div
  className="ui inverted icon input"
>
  <input
    onChange={[Function]}
    placeholder="test"
    type="text"
    value="a"
  />
  <i
    aria-hidden="true"
    className="search icon"
    onClick={[Function]}
  />
</div>
`;
